packages/compiler-cli/ngcc/src/host/umd_host.ts

```diff
@@ -521,10 +521,24 @@ export function parseStatementForUmdModule(statement: ts.Statement): UmdModule|n
   const factoryFn = stripParentheses(wrapper.call.arguments[factoryFnParamIndex]);
   if (!factoryFn || !ts.isFunctionExpression(factoryFn)) return null;
 
+  let factoryCalls: UmdModule['factoryCalls']|null = null;
+
   return {
     wrapperFn: wrapper.fn,
     factoryFn,
-    factoryCalls: parseUmdWrapperFunction(wrapper.fn),
+    // Compute `factoryCalls` lazily, because in some cases they might not be needed for the task at
+    // hand.
+    // For example, if we just want to determine if an entry-point is in CommonJS or UMD format,
+    // trying to parse the wrapper function could potentially throw a (premature) error. By making
+    // the computation of `factoryCalls` lazy, such as error would be thrown later (during an
```

Comment:
```suggestion
    // the computation of `factoryCalls` lazy, such an error would be thrown later (during an
```

Do we want to have a test for sniffing of UMD bundles that don't follow the expected structure?

Reply:
LoL - I had added a test, but apparently forgot to push it: https://github.com/angular/angular/pull/44245/commits/b926bc1fa24b221f108088dfd96f215e5031dd2b 😇